src/lib/legacy.ts

```diff
@@ -6531,6 +6531,52 @@ const toZigbee2 = {
             await sendDataPointRaw(entity, dataPoints.moesSchedule, payload);
         },
     } satisfies Tz.Converter,
+    moes_thermostat_program_schedule_v2: {
+        key: ['program_v2'],
+        convertSet: async (entity, key, value: any, meta) => {
+            if (!meta.state.program) {
+                logger.warning(`Existing program state not set.`, 'zhc:legacy:tz:moes_bht_002');
+                return;
+            }
+
+            /* Merge modified value into existing state and send all over in one go */
+            const newProgram = {
+                // @ts-expect-error ignore
+                ...meta.state.program,
+                ...value,
+            };
+
+            const getNumbers = (input: exposes.Text): number[] => {
+                input.checkPatternMatch();
```

Comment:
I don't understand what this does, how can `input` be `exposes.Text` here? Isn't it a string?

Reply:
aww
Sure. I've just started to dive into it and missed some core concepts. I've thought there is component and not clean result.

But maybe you'll suggest where can I make this check? I wanted to try to incapsulate all info inside component but not sure where this information can be used. As I see a lot of this stuff doing in z2m-frontend?

Reply:
I would you do a simple regex check here instead of ` input.checkPatternMatch();`

Reply:
I can but main idea was to improve something and provide kind of better incapsulated solution